example/src/main/java/org/wordpress/android/fluxc/example/ui/products/WooProductsFragment.kt

```
@@ -371,6 +377,26 @@ class WooProductsFragment : Fragment() {
             }
         }
 
+        fetch_product_addons.setOnClickListener {
+            selectedSite?.let { site ->
+                showSingleLineDialog(
+                        activity,
+                        "Enter the remoteProductId of product to fetch the addons:"
+                ) { editText ->
+                    pendingFetchSingleProductRemoteId = editText.text.toString().toLongOrNull()
+                    pendingFetchSingleProductRemoteId?.let { id ->
+                        prependToLog("Submitting request to fetch product by remoteProductID $id")
+                        coroutineScope.launch {
+                            wcProductStore.fetchProductListSynced(site, listOf(id))
+                                    .takeUnless { it.isNullOrEmpty() }
+                                    ?.first()?.addons
+                                    ?.logAddons()
+                        }
+                    } ?: prependToLog("No valid remoteOrderId defined...doing nothing")
```

❓ If we have a log for `not valid remoteOrderId`, maybe we could have one for not finding `addons`?

---

Actually, we do have a log for this scenario, but it's inside the `logAddons()` method, do you think it's too hidden and should be handled at the method caller instead of inside the method itself?

Here's how implemented:
```
private fun Array<WCProductAddonModel>?.logAddons() {
         this?.forEachIndexed { index, addon ->
             prependToLog(addon.description?.let { "description: $it" }.orEmpty())
             prependToLog(addon.name?.let { "name: $it" }.orEmpty())
             prependToLog("========== Product Add-on #$index =========")
         } ?: prependToLog("No addons found for this product ID")
     }
```

---

Oh missed that, thanks! But I think it might not work, please correct me if I'm wrong: 

if [`addons` are not found in payload, we return an emptyList](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2066/files#diff-62817f824846374ea4f2a7620ebef16d42efdb34f1990600f9c10b215c9c4f39R113). Next, in the snippet you've provided we log `No addons found for this product ID` for null `Array<WCProductAddonModel>` but it never happens.

I think `null`, if there are no add-ons in payload, is a valid state. I'd suggest to change `WCProductModel#addons` to return `null` if there are no add-ons, WDYT?

---

Great catch @wzieba! You're 100% correct, looks like I got fooled by my own safety measure 😅. I agree with your suggestion, it's way better to use a nullable state to represent data failure instead of an empty array. I've submitted the change right here: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2066/commits/65f612f39055b57ecae23a5ea8fd62141a9cff3e.

Let me know what you think and thanks for your time reviewing the PR!

---

Thanks for change! I've added one more comment here: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2066/files#r676488729